evidence/pool.go

```diff
@@ -144,14 +145,20 @@ func (evpool *Pool) IsCommitted(evidence types.Evidence) bool {
 	return ei.Evidence != nil && ei.Committed
 }
 
-func (evpool *Pool) removeEvidence(height, maxAge int64, blockEvidenceMap map[string]struct{}) {
+func (evpool *Pool) removeEvidence(
+	height int64,
+	lastBlockTime time.Time,
+	evidenceAge types.EvidenceParams,
+	blockEvidenceMap map[string]struct{}) {
 	for e := evpool.evidenceList.Front(); e != nil; e = e.Next() {
 		ev := e.Value.(types.Evidence)
 
+		evAge := lastBlockTime.Sub(ev.Time())
+
 		// Remove the evidence if it's already in a block
 		// or if it's now too old.
 		if _, ok := blockEvidenceMap[evMapKey(ev)]; ok ||
-			ev.Height() < height-maxAge {
+			ev.Height() < height-evidenceAge.MaxAgeNumBlocks || evAge > evidenceAge.MaxAgeDuration {
```

Comment:
```suggestion
			height > ev.Height()+evidenceAge.MaxAgeNumBlocks || evAge > evidenceAge.MaxAgeDuration {
```

better be consistent everywhere